src/test/java/tests/FilterEvalTests.java

```diff
@@ -130,6 +132,192 @@ public void milestone() {
+        msCurrMin3.setDueDate(Optional.of(LocalDate.now().minusMonths(2)));
+        TurboIssue iCurrMin3 = new TurboIssue(REPO, 2, "curr-3");
+        iCurrMin3.setMilestone(msCurrMin3);
+        iCurrMin3.setOpen(false);
+
+        // test: sort by due date is correct
+        TurboMilestone msCurrMin2 = new TurboMilestone(REPO, 9, "V0.2");
+        msCurrMin2.setOpen(false);
+        msCurrMin2.setDueDate(Optional.of(LocalDate.now().minusMonths(1)));
+        TurboIssue iCurrMin2 = new TurboIssue(REPO, 3, "curr-2");
+        iCurrMin2.setMilestone(msCurrMin2);
+
+        // test: future closed milestone will not be current milestone
+        TurboMilestone msCurrMin1 = new TurboMilestone(REPO, 8, "V0.3");
+        msCurrMin1.setOpen(false);
+        msCurrMin1.setDueDate(Optional.of(LocalDate.now().plusDays(1)));
+        TurboIssue iCurrMin1 = new TurboIssue(REPO, 4, "curr-1");
+        iCurrMin1.setMilestone(msCurrMin1);
+
+        // test: earliest future open milestone with 0 open issues will
+        // be current milestone
+        TurboMilestone msCurr = new TurboMilestone(REPO, 7, "V0.5");
+        msCurr.setOpen(true);
+        msCurr.setOpenIssues(0);
+        msCurr.setDueDate(Optional.of(LocalDate.now().plusMonths(1)));
+        TurboIssue iCurr = new TurboIssue(REPO, 5, "curr");
+        iCurr.setMilestone(msCurr);
+        iCurr.setOpen(false);
+
+        // test: sort by due date is correct, even if in the future but
+        // closed
+        TurboMilestone msCurrPlus1 = new TurboMilestone(REPO, 6, "V0.7");
+        msCurrPlus1.setOpen(false);
+        msCurrPlus1.setDueDate(Optional.of(LocalDate.now().plusMonths(2)));
+        TurboIssue iCurrPlus1 = new TurboIssue(REPO, 6, "curr+1");
+        iCurrPlus1.setMilestone(msCurrPlus1);
+
+        // test: sort by due date is correct
+        TurboMilestone msCurrPlus2 = new TurboMilestone(REPO, 5, "V0.8");
+        msCurrPlus2.setOpen(true);
+        msCurrPlus2.setDueDate(Optional.of(LocalDate.now().plusMonths(3)));
+        TurboIssue iCurrPlus2 = new TurboIssue(REPO, 7, "curr+2");
+        iCurrPlus2.setMilestone(msCurrPlus2);
+
+        // test: milestone with no due date should come last
+        TurboMilestone msCurrPlus3 = new TurboMilestone(REPO, 4, "V0.9");
+        msCurrPlus3.setDueDate(Optional.empty());
+        TurboIssue iCurrPlus3 = new TurboIssue(REPO, 8, "curr+3");
+        iCurrPlus3.setMilestone(msCurrPlus3);
+
+        model = TestUtils.singletonModel(new Model(REPO,
+                new ArrayList<>(Arrays.asList(iCurrMin3, iCurrMin2, iCurrMin1,
+                        iCurr, iCurrPlus1, iCurrPlus2, iCurrPlus3)),
+                new ArrayList<>(),
+                new ArrayList<>(Arrays.asList(msCurrMin3, msCurrMin2, msCurrMin1,
+                        msCurr, msCurrPlus1, msCurrPlus2, msCurrPlus3)),
+                new ArrayList<>()));
+
+        assertEquals(true, Qualifier.process(model, Parser.parse("milestone:curr-3"), iCurrMin3));
```

Comment:
Is this still a problem? I'm not sure why it would be immutable. Generally unit tests aren't as quirky as UI tests.

Tests for both would be good, with more emphasis on the former I guess.